File: src/BreakfastProvider.Api/Models/Requests/OrderRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record OrderRequest
{
    public string? CustomerName { get; init; }

    public List<OrderItemRequest> Items { get; init; } = [];

    public int? TableNumber { get; init; }
}

public record OrderItemRequest
{
    public string? ItemType { get; init; }

    public Guid? BatchId { get; init; }

    public int Quantity { get; init; } = 1;
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/OrderRequest.java
package io.lemonlion.breakfast.model.request;

import java.util.ArrayList;
import java.util.List;

/** Twin of C# {@code OrderRequest}. Validation is performed programmatically (see {@code OrderValidator}). */
public record OrderRequest(String customerName, List<OrderItemRequest> items, Integer tableNumber) {

    public OrderRequest {
        if (items == null) {
            items = new ArrayList<>();
        }
    }
}
